src/nested_pandas/nestedframe/core.py

```diff
@@ -219,10 +218,8 @@ def __setitem__(self, key, value):
             "." in key and key.split(".")[0] in self.nested_columns
         ):
             nested, col = key.split(".")
-            new_flat = self[nested].nest.to_flat()
-            new_flat[col] = value
-            packed = packer.pack(new_flat)
-            return super().__setitem__(nested, packed)
+            new_nested_series = self[nested].nest.with_flat_field(col, value)
```

Comment:
If I understand, this change avoids rewriting fields that haven't changed, true?

Reply:
Yes, and benchmarks show that it really helps